lib/spack/spack/package.py

```diff
@@ -1217,6 +1236,15 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class PythonExtension(Package):
+    def setup_dependent_environment(self, env, dependent_spec):
+        pass
+
+    def setup_environment(self, env):
+        site_packages = glob.glob(join_path(self.spec.prefix.lib, "python*/site-packages"))
+        if site_packages:
+            env.prepend_path('PYTHONPATH', site_packages[0])
```

Comment:
Forgot to mention that the other thing you miss with the subclass is that `setup_dependent_environment` is able to create the proper X.Y in `pythonX.Y/site-packages`, because it's called on the *extendee* (`python`).  So it knows the python version that the sub-package is extending, and can create the right path.  The version above uses glob and assumes someone has already created that directory, but I've seen `setuptools` do the wrong thing when `pythonX.Y/site-packages` doesn't exist in advance.  This is why it's done the way it is.

This is getting into a lot of python-specific stuff but I think the requirements and conventions are similarly complicated for other languages.  Maybe not quite as combinatorially versioned as python though...